DeviceAdministration/Infrastructure/Repository/DeviceRegistryRepository.cs

```diff
@@ -68,13 +69,14 @@ public DeviceRegistryRepository(IConfigurationProvider configProvider, IDocDbRes
                         true,
                         false) as IEnumerable;
 
-                deviceList = JsonConvert.DeserializeObject<List<DeviceModel>>(docs.ToString());
+                tmpDeviceList = JsonConvert.DeserializeObject<List<DeviceModel>>(docs.ToString());
+                deviceList.AddRange(tmpDeviceList);
```

Comment:
so, its a while loop running, we want to add more devices to deviceList in each iteration. If we initialize to deviceList, only the last iteration will be retained.